ml/kore/ast.py

```diff
@@ -516,6 +531,9 @@ def __eq__(self, other):
         if isinstance(other, Application):
             return self.symbol == other.symbol and self.arguments == other.arguments
         return False
+    
+    def __lt__(self, other) -> bool:
+        return str(self) < str(other)
```

Comment:
can we use a different order here? The `__str__` method is mostly used for debug and it may not be unique.
We could compare `arguments` and `symbol` using their overloaded comparison.